packages/@lwc/template-compiler/src/codegen/codegen.ts

```diff
@@ -392,17 +392,32 @@ export default class CodeGen {
     }
 
     genEventListeners(listeners: EventListener[]) {
-        const listenerObj = Object.fromEntries(
-            listeners.map((listener) => [listener.name, listener])
-        );
+        const listenerObj: Record<string, EventListener> = {};
+        for (const listener of listeners) {
+            listenerObj[listener.name] = listener;
+        }
+
```

Comment:
Object.fromEntries + map is known to be slower than for loop for constructing objects. This may help if there's a ton of listeners.

Reply:
Template compiler perf is not a huge concern right now, but I appreciate thinking about it!